### PR TITLE
Fixed #24986 -- Added support for annotations in DISTINCT queries

### PR DESCRIPTION
**Pull request by megaumi**
This pull request is split into two commits for easier review (the first one is a small fix for an inconsistency in some internal code). 

Fix for ticket https://code.djangoproject.com/ticket/24986.
Patch by Valentina Mukhamedzhanova and Donald Harvey @ Django sprint EuroPython 2015.

### CODE REVIEW

**Comment by timgraham** [comment]
buildbot, add to whitelist.

**Comment by timgraham** [comment]
Some cosmetic edits: http://dpaste.com/3C7SR6Q
Could you please rebase your branch and squash the commits (I merged the first one already).
Thanks!

**Comment by megaumi** [comment]
Done. Thank you, Tim!

File: django/db/models/sql/query.py [hunk 1]
```diff
@@ -1292,9 +1292,15 @@ def names_to_path(self, names, opts, allow_many=True, fail_on_missing=False):
             cur_names_with_path = (name, [])
             if name == 'pk':
                 name = opts.pk.name
+
+            field = None
             try:
                 field = opts.get_field(name)
+            except FieldDoesNotExist:
+                if name in self.annotation_select:
+                    field = self.annotation_select[name].field
```
Comment (jarshwah) [inline django/db/models/sql/query.py]: Please change to `field = self.annotation_select[name].output_field`. `output_field` is part of the expressions API. The field property is there for backwards compatibility with older code.

File: tests/annotations/tests.py [hunk 2]
```diff
@@ -160,6 +160,40 @@ def test_aggregate_over_annotation(self):
...
+        for name in names:
+            Employee.objects.create(
+                store=store,
+                first_name=name.split()[0],
+                last_name=name.split()[1],
+                age=30, salary=2000,
+            )
+
+        people = Employee.objects.annotate(
+            name_lower=Func(F('last_name'), function='lower'),
```
Comment (jarshwah) [inline tests/annotations/tests.py]: We could just use the `django.db.models.functions.Lower` function here, directly, without the Func wrapper. They produce exactly the same output though, so I'm not too concerned with this. Tim?

**Comment by jarshwah** [comment]
A couple of suggestions left above. Other than that LGTM.

File: tests/annotations/tests.py (hunk 2 above)
Comment (timgraham) [inline tests/annotations/tests.py]: using `Lower` seems more readable

**Comment by megaumi** [comment]
Updated the patch, thanks!

**Comment by timgraham** [comment]
merged in 1f7b25c1a7a85426675a04380f37b180edc08bbc, thanks!

**Pull request closed without merging**